hail/python/hail/backend/service_backend.py

```diff
@@ -8,14 +10,43 @@
 
 from hailtop.config import get_deploy_config, get_user_config
 from hailtop.auth import service_auth_headers
-from hailtop.utils import (retry_response_returning_functions,
-                           external_requests_client_session)
+from hailtop.utils import async_to_blocking
 from hail.ir.renderer import CSERenderer
 
 from .backend import Backend
 from ..hail_logging import PythonOnlyLogger
 
 
+class ServiceSocket:
+    def __init__(self, *, deploy_config=None):
+        if not deploy_config:
+            deploy_config = get_deploy_config()
+        self.url = deploy_config.base_url('query')
+        self.session = aiohttp.ClientSession(headers=service_auth_headers(deploy_config, 'query'))
+
+    def handle_response(self, resp):
+        if resp.type == aiohttp.WSMsgType.CLOSED:
+            raise FatalError('Socket was closed waiting for response from server.')
+        if resp.type == aiohttp.WSMsgType.ERROR:
+            raise FatalError(f'Error raised while waiting for response from server: {resp}.')
+        assert resp.type == aiohttp.WSMsgType.TEXT
+        return resp.data
+
+    async def async_request(self, endpoint, **data):
+        async with self.session.ws_connect(f'{self.url}/api/v1alpha/{endpoint}') as socket:
+            await socket.send_str(json.dumps(data))
+            resp = self.handle_response(await socket.receive())
+            if resp != endpoint:
+                raise FatalError(f'Error from server: {json.loads(resp)["error"]}')
```

Comment:
What function does this step of the protocol serve? IIUC, the protocol is:
```
CLIENT: ws://.../endpoint data
SERVER: endpoint
SERVER: data2
```

Reply:
whoops, yes. I'll probably just change this to `assert resp == endpoint`, I think I got it mixed up with the next response.

Reply:
I still think we're sending an unnecessary message. Why does query respond with the endpoint? The client knows the endpoint (it requested it).